services/spring-boot-starter-datawave-audit/src/main/java/datawave/microservice/audit/AuditClient.java

```diff
@@ -0,0 +1,208 @@
+
+import com.google.common.base.Preconditions;
+import datawave.marking.SecurityMarking;
+import datawave.microservice.authorization.jwt.JWTRestTemplate;
+import datawave.microservice.authorization.user.ProxiedUserDetails;
+import datawave.security.authorization.DatawaveUser;
+import datawave.webservice.common.audit.AuditParameters;
+import datawave.webservice.common.audit.Auditor;
+import datawave.webservice.common.audit.Auditor.AuditType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * Simple rest client for submitting requests to the audit service
+ *
+ * @see Request
+ * @see AuditServiceProvider
+ */
+@Service
+@ConditionalOnProperty(name = "audit-client.enabled", havingValue = "true", matchIfMissing = true)
+public class AuditClient {
+    
+    private static final String DEFAULT_REQUEST_PATH = "/v1/audit";
+    
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    private final AuditServiceProvider serviceProvider;
+    private final JWTRestTemplate jwtRestTemplate;
+    
+    @Autowired
+    public AuditClient(RestTemplateBuilder builder, AuditServiceProvider serviceProvider) {
+        this.jwtRestTemplate = builder.build(JWTRestTemplate.class);
+        this.serviceProvider = serviceProvider;
+    }
+    
+    public void submit(Request request) {
+        submit(request, DEFAULT_REQUEST_PATH);
+    }
+    
+    public void submit(Request request, String requestPath) {
+        
+        Preconditions.checkNotNull(request, "request cannot be null");
+        Preconditions.checkNotNull(requestPath, "requestPath cannot be null");
+        
+        log.debug("Submitting audit request: {}", request.getAuditParameters());
+        
+        if (AuditType.NONE.equals(request.getAuditType())) {
+            if (serviceProvider.getProperties().isSuppressAuditTypeNone()) {
+                log.debug("Audit request with AuditType == {} was suppressed", AuditType.NONE);
+                return;
+            }
+        }
+        
+        //@formatter:off
+        ServiceInstance auditService = serviceProvider.getServiceInstance();
+        UriComponents uri = UriComponentsBuilder.fromUri(auditService.getUri())
+            .path(auditService.getServiceId() + requestPath)
+            .queryParams(request.getAuditParametersAsMap())
```

Comment:
The audit request is a POST request, and the params should be passed in the body rather than as query parameters.

Reply:
Still working this one. Wasn't difficult to fix here, but it affects a bunch of tests since it changes the uri assertions in all my audit-related junits

Reply:
Fixed in latest commit